File: cdev/src/modules/users/application/users/get-current-user/get-current-user.query-handler.ts
import { QueryHandler } from '@cdev/common/application';
import { User } from '@cdev/modules/users/domain';
import { UserRepository } from '../../abstractions/user-repository';
import { GetCurrentUserQuery } from './get-current-user.query';

export class GetCurrentUserQueryHandler implements QueryHandler<GetCurrentUserQuery, User> {
  constructor(private readonly repository: UserRepository) {}

  handle(_query: GetCurrentUserQuery): Promise<User> {
    return this.repository.getCurrentUser();
  }
}

File: api/src/Modules/Users/Application/Users/GetCurrentUser/GetCurrentUserQueryHandler.cs
namespace Checknote.Modules.Users.Application.Users.GetCurrentUser;

using System.Threading;
using System.Threading.Tasks;
using Checknote.Common.Application.Messaging;
using Checknote.Common.Domain;
using Checknote.Modules.Users.Application.Abstractions;
using Checknote.Modules.Users.Domain.Users;

public sealed class GetCurrentUserQueryHandler : IQueryHandler<GetCurrentUserQuery, User>
{
    private readonly ICurrentUserProvider currentUserProvider;

    public GetCurrentUserQueryHandler(ICurrentUserProvider currentUserProvider)
    {
        this.currentUserProvider = currentUserProvider;
    }

    public Task<Result<User>> Handle(GetCurrentUserQuery query, CancellationToken cancellationToken)
    {
        return Task.FromResult(currentUserProvider.GetCurrentUser());
    }
}
